web-ui/src/components/certifications/EarnedCertificationsTable.jsx

```diff
@@ -0,0 +1,566 @@
+import { format } from 'date-fns';
+import PropTypes from 'prop-types';
+import React, { useCallback, useContext, useEffect, useState } from 'react';
+
+import {
+  AddCircleOutline,
+  Delete,
+  Edit,
+  EmojiEvents
+} from '@mui/icons-material';
+import {
+  Autocomplete,
+  Button,
+  Card,
+  CardContent,
+  CardHeader,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  IconButton,
+  TextField,
+  Tooltip
+} from '@mui/material';
+
+import DatePickerField from '../date-picker-field/DatePickerField';
+import ConfirmationDialog from '../dialogs/ConfirmationDialog';
+import { AppContext } from '../../context/AppContext';
+import { selectCurrentUser, selectProfileMap } from '../../context/selectors';
+import './EarnedCertificationsTable.css';
+
+const certificationBaseUrl = 'http://localhost:3000/certification';
+const earnedCertificationBaseUrl = 'http://localhost:3000/earned-certification';
+
+const formatDate = date => {
+  return !date
+    ? ''
+    : date instanceof Date
+      ? format(date, 'yyyy-MM-dd')
+      : `${date.$y}-${date.$M + 1}-${date.$D}`;
+};
+
+const newEarned = { earnedDate: formatDate(new Date()) };
+const tableColumns = [
+  'Member',
+  'Name',
+  'Description',
+  'Earned On',
+  'Expiration',
+  'Certificate Image',
+  'Badge'
+];
+
+const propTypes = {
+  forceUpdate: PropTypes.func,
+  onlyMe: PropTypes.bool
+};
+
+const EarnedCertificationsTable = ({
+  forceUpdate = () => {},
+  onlyMe = false
```

Comment:
We pass `true` for `onlyMe` when this component is used in `MemberProfilePage.jsx`. In that case we hide the "Member" column and only display certifications earned by the currently logged in user. They can only create, edit, and delete their own earned certifications.